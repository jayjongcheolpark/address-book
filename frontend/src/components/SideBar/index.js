import React from 'react';
import styled from 'styled-components';

import SearchBar from '../SearchBar';
import ContactList from '../Contact/List';

const ColorDiv = styled.div`
  background-color: #7e57c2;
  padding: 10px;
  color: white;
`;

const CenterDiv = styled.div`
  text-align: center;
`;

const SideBar = () => (
  <ColorDiv>
    <CenterDiv>
      <h2>All Contacts</h2>
    </CenterDiv>
    <SearchBar />
    <ContactList />
  </ColorDiv>
);

export default SideBar;
