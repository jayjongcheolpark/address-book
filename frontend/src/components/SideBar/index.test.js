import React from 'react';
import { shallow } from 'enzyme';

import SideBar from './index';

describe('App', () => {
  const sideBar = shallow(<SideBar />);

  it('renders properly', () => {
    expect(sideBar).toMatchSnapshot();
  });

  it('contains a connected SearchBar', () => {
    expect(sideBar.find('Connect(SearchBar)').exists()).toBe(true);
  });
  it('contains a connected ContactList', () => {
    expect(sideBar.find('Connect(Apollo(ContactList))').exists()).toBe(true);
  });
});
