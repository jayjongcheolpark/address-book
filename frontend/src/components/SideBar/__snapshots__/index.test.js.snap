// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <h2>
                    All Contacts
          </h2>
</styled.div>,
        <Connect(SearchBar) />,
        <Connect(Apollo(ContactList)) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h2>
            All Contacts
</h2>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "All Contacts",
          },
          "ref": null,
          "rendered": "All Contacts",
          "type": "h2",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <h2>
                        All Contacts
            </h2>
</styled.div>,
          <Connect(SearchBar) />,
          <Connect(Apollo(ContactList)) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h2>
              All Contacts
</h2>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "All Contacts",
            },
            "ref": null,
            "rendered": "All Contacts",
            "type": "h2",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
